lua/autorun/server/sv_daily_restart.lua

```diff
@@ -1,67 +1,88 @@
-        -- 15 min, 10min, 5min, 4min, 3min,2min,1min,30s,15s,10,9,8.. so on
-        sendAlertToClients()
+    --local params = {}
+    --local headers = {}
+    --http.Post("localhost:2327/gmod/restart", params, function() end, function() end, headers)
+end
 
-        net.Start("AlertUsersOfRestart")
-            net.WriteFloat(restartAt)
-        net.Broadcast()
+local function CFCDailyRestart:sendAlertToClients(message)
+    for k, v in pairs(player.GetHumans()) do
+        v:ChatPrint("[CFC Daily Restart] "..message)
     end
+end
+
+local function CFCDailyRestart:alertClientsOfServerRestart()
+    print("Alerting clients of Restart")
+    CFCDailyRestart.restartAt = os.time() + (warningTime * 60)
+
+    CFCDailyRestart:sendAlertToClients("Restarting server in "..CFCDailyRestart.warningTime.." minutes.")
+
+    net.Start("AlertUsersOfRestart")
+        net.WriteFloat(restartAt)
+    net.Broadcast()
+end
 
-    --TODO: Create a timer to 
+timer.Create("CFC_DailyRestart", 1, 0, function()
+    local currentTime = os.time()
+    local lastScheduledRestart = CFCDailyRestart.lastRestart
+    local isLastScheduledRestartValid = not lastScheduledRestart == nil
+    local scheduledTimeThreshold = 3600 --3600 = 1 hour
 
-    timer.Create("CFC_DailyRestart", 1, 0, function()
-        -- TODO: Check to make sure that the latest restart time wasn't recently
-        -- if countingDownToRestart then return
-        -- if waitingForServerToEmpty then return
+    if isLastScheduledRestartValid then
+        if os.difftime(os.time(), lastScheduledRestart) <= scheduledTimeThreshold then return end
+    end
+
+    local formattedTime = os.date("*t", currentTime)
 
-        theTime = os.date("*t", os.time())
+    -- TODO: Also check each second within 1 hour leading up to the restart time. If the server is empty during this time, issue the restart and ensure we don't try to restart again at the desired time
+    --
+    -- TODO (Future): Perhaps if there are fewer than N players on the server leading up to the restart time, allow them to vote to restart now and get it out of the way
 
-        -- TODO: Also check each second within 1 hour leading up to the restart time. If the server is empty during this time, issue the restart and ensure we don't try to restart again at the desired time
-        --
-        -- TODO (Future): Perhaps if there are fewer than N players on the server leading up to the restart time, allow them to vote to restart now and get it out of the way
+    if not countingDownToRestart then
+        local isHourEquivalentToDesired = (formattedTime.hour == desiredRestartTime)
+        local secondThreshold = 30
+        local isSecondWithinThreshold = (formattedTime.sec <= secondThreshold)
 
-        if (theTime.hour == desiredRestartTime) and (theTime.sec <= 30) then
+        if isHourEquivalentToDesired and isSecondWithinThreshold then
             local plyCount = table.Count(player.GetHumans())
 
             if plyCount == 0 then
```

Comment:
Something that I like to do is avoid comparisons in conditionals.

Instead, I might write:

```lua
local playerCount = table.Count(player.GetHumans())
local serverIsEmpty = playerCount == 0

if serverIsEmpty:
```